backend/src/main/java/com/eventus/backend/controllers/ParticipationController.java

```diff
@@ -0,0 +1,97 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+@RestController
+public class ParticipationController {
+
+    private final ParticipationService participationService;
+
+    private final UserService userService;
+
+    private final EventService eventService;
+
+    @Autowired
+    public ParticipationController(ParticipationService participationService,UserService userService,EventService eventService) {
+        this.participationService = participationService;
+        this.userService = userService;
+        this.eventService=eventService;
+    }
+
+    @GetMapping("/participations")
+    @ResponseStatus(HttpStatus.OK)
+    public List<Participation> getParticipations(@RequestParam(defaultValue = "0") Integer numPag) {
+        return this.participationService.findAllParticipation(PageRequest.of(numPag,20));
+    }
+
+    @GetMapping("/participations/{id}")
+    public ResponseEntity<Participation> getParticipationById(@PathVariable Long id) {
+        Participation participation=this.participationService.findParticipationById(id).orElse(null);
+        if(participation!=null){
+            return ResponseEntity.ok(participation);
+        }else{
+            return ResponseEntity.notFound().build();
+        }
+
+    }
+
+    @GetMapping("/user/event/{eventId}")
+    public ResponseEntity<List<User>> getUsersByEvent(@PathVariable Long eventId, @RequestParam(defaultValue = "0") Integer numPag) {
+       return ResponseEntity.ok(this.participationService.findUsersByEventId(eventId, PageRequest.of(numPag,20)));
+    }
+
+    @GetMapping("/participations/user/{userId}")
+    public ResponseEntity<List<Participation>>  getParticipationsByUser(@PathVariable Long userId, @RequestParam(defaultValue = "0") Integer numPag) {
+        return ResponseEntity.ok(this.participationService.findParticipationByUserId(userId,PageRequest.of(numPag,20)));
+    }
+
+    @GetMapping("/participations/event/{eventId}")
+    public ResponseEntity<List<Participation>> getParticipationsByEvent(@PathVariable Long eventId, @RequestParam(defaultValue = "0") Integer numPag) {
+        return ResponseEntity.ok(this.participationService.findParticipationByEventId(eventId, PageRequest.of(numPag,20)));
+    }
+
+    @PostMapping("/participations")
+    public ResponseEntity<String> createParticipation(@RequestBody Map<String,String> p) {
+        try{
+            Event event= this.eventService.findById(Long.valueOf(p.get("eventId")));
+            User user = this.userService.findUserById(1L).orElse(null);
+            if(user!=null&&event!=null){
+                Participation participation=new Participation();
+                participation.setTicket(RandomStringUtils.randomAlphanumeric(20));
+                participation.setPrice(event.getPrice());
+                participation.setEvent(event);
+                participation.setUser(user);
+                this.participationService.saveParticipation(participation);
+            }else{
```

Comment:
Creation of the participation should be managed by the service. This code should be implemented in its appropiate service (ParticipationService). It would be great if in this method just called the service like: 
'''
this.eventService.createParticipation(event) 
'''

Reply:
The creation of the participation has been move to userService